qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -12,19 +12,32 @@
 
 """A standard gradient descent optimizer."""
 
-from typing import Iterator, Optional, Union, Callable, Dict, Any, List, Tuple
-from functools import partial
-
+from dataclasses import dataclass
+from typing import Dict, Any, Union, Callable, Optional, Tuple, List, Iterator
 import numpy as np
-
 from .optimizer import Optimizer, OptimizerSupportLevel, OptimizerResult, POINT
+from .steppable_optimizer import AskObject, TellObject, OptimizerState, SteppableOptimizer
 
 CALLBACK = Callable[[int, np.ndarray, float, float], None]
 
 
-class GradientDescent(Optimizer):
-    r"""The gradient descent minimization routine.
+def constant(eta=0.01):
```

Comment:
This function exists just so that in case we want to use a constant learning rate we don't need to create a generator ourselves. I will document it a little bit and maybe change the name to ``constant_generator_factory``.

Reply:
This function already existed, right? In that case I don't think we have to change anything 😄 (unless we add proper deprecation etc, but this is mostly an internal function)